File: quest/DSA/Array1/array1.py
from typing import List

class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        # len(nums) returns the number of elements in the list (its length).
        n = len(nums)

        # Create a list of size 2*n filled with zeros.
        # Syntax: [value] * k repeats "value" k times.
        ans = [0] * (2 * n)

        # range(n) generates numbers from 0 to n-1 (0-indexed loop).
        for i in range(n):
            # Copy original element into the first half.
            ans[i] = nums[i]

            # Copy the same element into the second half (shifted by +n).
            ans[i + n] = nums[i]

        return ans

File: quest/DSA/Array1/array1.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        // nums.size() returns the number of elements in the vector.
        int n = nums.size();

        // Create a vector of size 2*n filled with 0.
        vector<int> ans(2 * n, 0);

        // Loop from 0 to n-1.
        for (int i = 0; i < n; i++) {
            // Copy original element into the first half.
            ans[i] = nums[i];

            // Copy the same element into the second half (shifted by +n).
            ans[i + n] = nums[i];
        }

        return ans;
    }
};
